src/test/java/me/karakelley/http/publicdirectory/PublicDirectoryTest.java

```diff
@@ -197,4 +187,19 @@ void testUpdateContentsOfFile() {
     });
   }
 
+  @Test
+  void testValidPath() {
+    TempFilesHelper.withTempDirectory(directory ->  {
+      PublicDirectory publicDirectory = PublicDirectory.create(directory.toString(), new FileFinderCache(new RealFileFinder()));
+      assertEquals(true, publicDirectory.isValidPath("testing/../testing"));
+    });
+  }
+
+  @Test
+  void testInvalidPath() {
```

Comment:
Follow-up on the public-vs-private `isValidPath` comment:
On the one hand, I like these two tests here because they help document the behavior of `isValidPath` method, which I personally find a little hard to follow (but probably mainly because it uses a lot of std lib methods with which I'm not super familiar). On the other hand, the 404-on-attempted-dir-traversal test does cover the public-facing behavior at a nice, high level. Maybe there's a compromise, keeping the 404 test and replacing these with something just a little higher up in `PublicDirectory`?

Reply:
The method should be private. I initially had a different implementation and when I changed it to be called within `resourceExists()` instead I forgot to change it to private. I could easily test this through `resourceExists()` which is probably more appropriate entry point for testing it.